opensquirrel/replacer.py

```diff
@@ -41,6 +42,10 @@ def visit_controlled_gate(self, controlled_gate: ControlledGate):
         result = ControlledGate(control_qubit=control_qubit, target_gate=target_gate)
         return result
 
+    def visit_measurement_operation(self, m: Measure):
+        result = Measure(qubit=Qubit(self.mappings.index(m.qubit)), axis=m.axis)
+        return result
```

Comment:
The `_QubitReIndexer` only works on gates, so I'm not sure if this should be here. The `decompose` method skips anything other than `Gate`s, so it's not needed.

We could throw an error if the user wants to use the `Circuit.replace` method to replace a measure operation with a list of gates.

Reply:
The error (i.e. that a measure instruction cannot be replaced by a list of gates) shouldn't be part of a visitor method of the `_QubitReIndexer`, it should be part of the `replace` function in `replacer.py`.